### ISSUE TITLE
Required but nullable object field is non-nullable in generated class 

### ISSUE THREAD

**Issue by DavidMazarro** [cjbooms/fabrikt#289]
When running a schema like this through Fabrikt:
```yaml
components:
  schemas:
    MyObject:
      type: object
      required:
        - foo
      properties:
        foo:
          type: object
          nullable: true
```
The generated class looks like this:
```kotlin
public data class MyObject(
  @param:JsonProperty("foo")
  @get:JsonProperty("foo")
  @get:NotNull
  public val foo: Map<String, Any>,
)
```
But if I'm not mistaken, one would expect:
```kotlin
public data class MyObject(
  @param:JsonProperty("foo")
  @get:JsonProperty("foo")
  @get:NotNull
  public val foo: Map<String, Any?>,
)
```

**Comment by wduncanfraser** [comment]
Seeing similar behavior with an array of nullable strings. For example:

```
        ips:
          type: array
          items:
            type: string
            nullable: true

```

Results in `List<String>` instead of `List<String?>`.

**Comment by cjbooms** [comment]
Thanks for the report. 

Support for `nullable: true` was only a recent addition, so we probably missed some spots. 
https://github.com/cjbooms/fabrikt/pull/265

Are either of you willing to take a pop at fixing it? I'm not sure how soon before I can get around to fixing myself, but can support with review and release of fix rapidly.

**Comment by DavidMazarro** [comment]
Thanks for the quick response @cjbooms!

I'm going to give this a shot. I'll open a PR if I can get the fix ready along with some tests.

**Comment by DavidMazarro** [comment]
Fix for this was included in release 16.3.0.
